src/components/ReportActionItem/MoneyRequestView.tsx

```diff
@@ -390,9 +391,10 @@ function MoneyRequestView({
             {shouldShowAnimatedBackground && <AnimatedEmptyStateBackground />}
             <>
                 {shouldShowReceiptHeader && (
-                    <ReceiptAuditHeader
+                    <ReceiptAudit
                         notes={receiptViolations}
-                        shouldShowAuditMessage={!!(shouldShowNotesViolations && didRceiptScanSucceed)}
+                        shouldShowAuditSuccess={shouldShowAuditMessage && didRceiptScanSucceed}
+                        shouldShowAuditFailure={shouldShowAuditMessage && hasReceipt}
```

Comment:
I've been looking at this for a minute, and it's not obvious to me why we now need shouldShowAuditSuccess and shouldShowAuditFailure, instead of using notes.length to determine what to show.

What cases would we miss if we went back to relying only on notes.length? If you can update the docs for these params to make it clear, please do

Reply:
Can you confirm, and if you do, add a manual tests that would reproduce the issue, and confirm your change solves it. If you can't confirm, let's revert to what we had previously

Reply:
@Krishna2323 sorry if this is obvious, but can you explain why we can't use a single prop? Both success and failure need to be shown only if there's a receipt, and the issue you linked shows an error when there's no receipt. What am I missing? Can you share test steps to reproduce the issue when we use only 1 prop, that doesn't happen with this solution?

Reply:
I think I get it now after another look at the code, sorry about that. The thing is we want to show
1. Nothing if there's no receipt and one can't be added
2. "Receipt" if there is no receipt, but one can be added (receipt empty state would show below this)
3. "Receipt * Issue found" if there's a receipt, and there are notes
4. "Receipt * Verified" if there's a receipt, and no notes

So what about doing something like below, and we get rid of `shouldShowAuditSuccess` & `shouldShowAuditFailure`?

```
const shouldShowReceiptAudit = isReceiptAllowed && (shouldShowReceiptEmptyState || hasReceipt);

// New
// Set receiptViolations to empty so no audit messages are shown
if (!shouldShowAuditMessage || !didRceiptScanSucceed) {
    receiptViolations = [];
}
```

Then for the cases above
1. We'd show nothing if there's no receipt and one can't be added based on `{shouldShowReceiptAudit &&<ReceiptAudit ...` and the fact that `shouldShowReceiptAudit` would be false
2. "Receipt" since `shouldShowReceiptAudit` would be true and the ReceiptAudit component always renders "Receipt", and we'd set `receiptViolations` (`notes` inside the component) to empty array because there's no receipt and `didRceiptScanSucceed` would be false
3. "Receipt * Issue found" since `shouldShowReceiptAudit` would be true, and  both `shouldShowAuditMessage` and `didRceiptScanSucceed` would be true, so we won't set `receiptViolations` to empty array
4. "Receipt * Verified" since `shouldShowReceiptAudit` would be true, and `receiptViolations`/`notes` will already be empty

Reply:
Love it, sorry for the late reply

Reply:
@Krishna2323 will you be able to update this today and get the PR ready for review?